testsuite/tests/typing-local/local.ml

```diff
@@ -1228,6 +1228,8 @@ let foo y =
   gbl
 [%%expect{|
 val foo : 'a -> 'a = <fun>
+|}, Principal{|
+val foo : '_weak1 -> '_weak1 = <fun>
```

Comment:
is this ok? what's going on here?

Reply:
Don't know and don't care. We don't use non-principal mode within Jane Street. (Other than within the compiler.) But we should definitely leave a CR to come back here when we take a run at fixing principal-mode problems.

Reply:
sgtm